### PR TITLE
feat(FN-1725): commonise some facility constants

### PR DESCRIPTION
**Pull request by JoshBinns2000**
## Introduction :pencil2:
We want to commonise the `FACILITY_TYPE` and `FACILITY_PROVIDED_DETAILS` constant.

## Resolution :heavy_check_mark:
- Moves the constants to `libs/common`
- Creates a new `FacilityType` type
- Updates the `Facility` db model type
- Updates references to these constants across services

### CODE REVIEW

File: utils/data-migration/gef/migration-map.js [hunk 1]
```diff
@@ -73,17 +74,17 @@ const DEAL = {
...
 };
 
 const BASIS_DETAILS = {
-  'a facility provided on a term basis': V2_CONSTANTS.FACILITIES.GEF_FACILITY_PROVIDED_DETAILS.TERM,
-  'a facility provided on a revolving or renewing basis': V2_CONSTANTS.FACILITIES.GEF_FACILITY_PROVIDED_DETAILS.RESOLVING,
-  'a facility provided on a committed basis': V2_CONSTANTS.FACILITIES.GEF_FACILITY_PROVIDED_DETAILS.COMMITTED,
-  'a facility provided on an uncommitted basis': V2_CONSTANTS.FACILITIES.GEF_FACILITY_PROVIDED_DETAILS.UNCOMMITTED,
-  'a facility provided on an on-demand or overdraft basis': V2_CONSTANTS.FACILITIES.GEF_FACILITY_PROVIDED_DETAILS.ON_DEMAND,
-  'a factoring facility provided on a with-recourse basis': V2_CONSTANTS.FACILITIES.GEF_FACILITY_PROVIDED_DETAILS.FACTORING,
+  'a facility provided on a term basis': FACILITY_PROVIDED_DETAILS.TERM,
```
Comment (AlexBramhill) [inline utils/data-migration/gef/migration-map.js:82]: sorry -- late night review questions. What on earth is a V2 constant?

File: libs/common/src/constants/facility-type.ts [hunk 2]
```diff
@@ -1,8 +1,6 @@
-const FACILITY_TYPE = {
+export const FACILITY_TYPE = {
```
Comment (AlexBramhill) [inline libs/common/src/constants/facility-type.ts:1]: just a comment to say how fancy is github realising this file has moved and displaying it nicely

File: dtfs-central-api/api-tests/mocks/gef/gef-facilities.ts [hunk 3]
```diff
@@ -1,47 +1,36 @@
...
-  detailsOther: null,
-  currency: null,
-  value: null,
-  coverPercentage: null,
-  interestPercentage: null,
-  paymentType: null,
-};
-
-const mockFacilities = [
+export const mockFacilities: Facility[] = [
```
Comment (AlexBramhill) [inline dtfs-central-api/api-tests/mocks/gef/gef-facilities.ts:5]: nice job with this even though not needed as part of this review

**Review by AlexBramhill — APPROVED** [review]
Really nice job here, and good job keeping the scope tight to make it easier to review!

File: utils/data-migration/gef/migration-map.js (hunk 1 above)
Comment (JoshBinns2000) [reply utils/data-migration/gef/migration-map.js:82]: I have no idea 😅

File: libs/common/src/constants/facility-type.ts (hunk 2 above)
Comment (JoshBinns2000) [reply libs/common/src/constants/facility-type.ts:1]: I thought the same!

File: dtfs-central-api/api-tests/mocks/gef/gef-facilities.ts (hunk 3 above)
Comment (JoshBinns2000) [reply dtfs-central-api/api-tests/mocks/gef/gef-facilities.ts:5]: I had already added the `aFacility` test helper in another branch but it made sense to move it here

**Pull request merged**